examples/controls_example/public/react_controls/data_controls/initialize_data_control.tsx

```diff
@@ -109,15 +109,44 @@ export const initializeDataControl = <EditorState extends object = {}>(
   );
 
   const onEdit = async () => {
-    openDataControlEditor<DefaultDataControlState & EditorState>(
-      { ...stateManager, ...editorStateManager } as ControlStateManager<
-        DefaultDataControlState & EditorState
-      >,
-      controlGroup,
+    // get the initial state from the state manager
+    const mergedStateManager = {
+      ...stateManager,
+      ...editorStateManager,
+    } as ControlStateManager<DefaultDataControlState & EditorState>;
+    const initialState = (
+      Object.keys(mergedStateManager) as Array<keyof DefaultDataControlState & EditorState>
+    ).reduce((prev, key) => {
+      return {
+        ...prev,
+        [key]: mergedStateManager[key]?.getValue(),
+      };
+    }, {} as DefaultDataControlState & EditorState);
+
+    // open the editor to get the new state
+    openDataControlEditor<DefaultDataControlState & EditorState>({
       services,
-      controlType,
-      controlId
-    );
+      onSave: ({ type: newType, state: newState }) => {
+        if (newType === controlType) {
+          // apply the changes from the new state via the state manager
+          (Object.keys(initialState) as Array<keyof DefaultDataControlState & EditorState>).forEach(
+            (key) => {
+              mergedStateManager[key].next(newState[key]);
```

Comment:
Should there be a check to only fire `next` if the value has changed? As written, this will fire `next` for each key in state even when the value has not changed. For example, dataViewId, will fire even when the same, causing a control to re-fetch the data view. Consumers of PublishingSubjects should not be in charge of ensuring distinct.

Reply:
Good call - done in 3bf7641cd4562cbb914a22f8caa43e8ff6cf5818 :+1: